### PR TITLE
Add a doc for the garbage collector

### PR DESCRIPTION
**Pull request by caesarxuchao**
cc @lavalamp @johndmulhausen 

@pwittrock do you know how to add the doc to the navigation sidebar?

### CODE REVIEW

**Comment by johndmulhausen** [comment]
http://kubernetes.io/editdocs/ has instructions on left nav editing.

File: docs/user-guide/garbage-collector.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+* TOC
+{:toc}
+
+## WARNING: Garbage Collector is an alpha feature, use it at your own risk!
+
+### What does the garbage collector do
+
+The garbage collector (GC) monitors the cluster and checks the metadata.ownerReferences field of each object. If all the owners present in metadata.ownerReferences do not exist in the cluster, the GC will request the API server to delete the object.
```
Comment (johndmulhausen) [inline docs/user-guide/garbage-collector.md]: Surround metadata.ownerReferences in backtics `like so`

Comment (johndmulhausen) [inline docs/user-guide/garbage-collector.md]: If any the owners present in metadata.ownerReferences do not exist in the cluster, the GC will delete the object.

File: docs/user-guide/garbage-collector.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+When deleting an object, if you want to prevent the GC from deleting objects that depend on it, you can send the deletion request with `deleteOptions.orphanDependents=true`, then the GC will remove the object from its dependents' metadata.ownerReferences, so that the dependents will not be garbage collected.
```
Comment (johndmulhausen) [inline docs/user-guide/garbage-collector.md]: When deleting an object, you can prevent the GC from deleting that object's dependents by specifying `deleteOptions.orphanDependents=true` in the deletion request. This flag prevents garbage collection by removing the deleted object from its dependents' `metadata.ownerReferences` field.

File: docs/user-guide/garbage-collector.md (hunk 1 above)
Comment (caesarxuchao) [inline docs/user-guide/garbage-collector.md]: Thanks. I'll reword it as "If none of the owners present in metadata.ownerReferences exists in the cluster, the GC will delete the object."

"If any of the owners..." is not accurate.

**Comment by caesarxuchao** [comment]
Addressed the comments and updated the nav bar.

File: _data/guides.yml [hunk 3]
```diff
@@ -58,6 +58,8 @@ toc:
     path: /docs/user-guide/config-best-practices/
   - title: Using kubectl to Manage Resources
     path: /docs/user-guide/working-with-resources/
+  - title: Garbage collection
```
Comment (pwittrock) [inline _data/guides.yml]: We also have Garbage collection for container images.  Is a way to disambiguate?

File: docs/user-guide/garbage-collector.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+---
+
+* TOC
+{:toc}
+
+## WARNING: Garbage Collector is an alpha feature, use it at your own risk!
```
Comment (pwittrock) [inline docs/user-guide/garbage-collector.md]: Is it enabled by default?

Comment (pwittrock) [inline docs/user-guide/garbage-collector.md]: Reading on it is not.  Best to start with this when talking about alpha.

File: docs/user-guide/garbage-collector.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+---
+---
+
+* TOC
+{:toc}
+
+## WARNING: Garbage Collector is an alpha feature, use it at your own risk!
+
+### What does the garbage collector do
```
Comment (pwittrock) [inline docs/user-guide/garbage-collector.md]: This section has a lot of implementation detail.  Start with what is meaningful to the user, and put implementation detail in a separate section.  What is the Purpose? - e.g. to delete resources created by delete controllers.

File: docs/user-guide/garbage-collector.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+* TOC
+{:toc}
+
+## WARNING: Garbage Collector is an alpha feature, use it at your own risk!
+
+### What does the garbage collector do
+
+The garbage collector (GC) monitors the cluster and checks the `metadata.ownerReferences` field of each object. If none of the owners present in `metadata.ownerReferences` exists in the cluster, the GC will request the API server to delete the object.
```
Comment (pwittrock) [inline docs/user-guide/garbage-collector.md]: When are these created?  What are these created on?  Where are they created (client-side, server side?)  Do these already exist, or are the new in 1.3?  Are these created only if GC is turned on?  How do I get GC for controllers that already exist?

File: docs/user-guide/garbage-collector.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (pwittrock) [inline docs/user-guide/garbage-collector.md]: How long does GC take?  Is there any way to be notified when GC is complete?

File: _data/guides.yml (hunk 3 above)
Comment (caesarxuchao) [inline _data/guides.yml]: Thanks Phil. I'll get back to your comments tomorrow.

**Comment by pwittrock** [comment]
@caesarxuchao Ping

File: _data/guides.yml (hunk 3 above)
Comment (caesarxuchao) [inline _data/guides.yml]: Changed the title of that one to kubelet garbage collection.

File: docs/user-guide/garbage-collector.md (hunk 7 above)
Comment (caesarxuchao) [inline docs/user-guide/garbage-collector.md]: > How long does GC take?

We don't have numbers yet.

>  Is there any way to be notified when GC is complete?
No, there is no special notification mechanism yet. We have thought about this and have found solutions, but I don't want to mention it in the doc as we don't have a timeline yet.

File: docs/user-guide/garbage-collector.md (hunk 6 above)
Comment (caesarxuchao) [inline docs/user-guide/garbage-collector.md]: `metadata.ownerReferences` needs to be manually created in 1.3 but will be automatically created in 1.4. I moved the `Work In Progress` paragraph to here so hopefully readers will get answers to these questions.

**Comment by caesarxuchao** [comment]
@pwittrock I pushed another version. PTAL. Thanks!

**Comment by johndmulhausen** [comment]
Thanks for the tech review @pwittrock and thanks for the PR @caesarxuchao ! Merging so we can see this staged; passes the test structurally. :)

**Pull request merged**